spec/controllers/groups_controller_spec.rb

```diff
@@ -241,6 +241,53 @@
       end
     end
 
+    describe '#create_groups_when_students_work_alone' do
+      before :each do
+        @assignment = create(:assignment)
+        # Create students
+        @student_user_names = %w[c8shosta c5bennet]
+        @student_user_names.each do |name|
+          create(:student, user: create(:end_user, user_name: name))
+        end
+      end
+
+      it 'creates groups for individual students' do
+        data = [%w[c8shosta c8shosta], %w[c5bennet c5bennet]]
+        ActiveJob::Base.queue_adapter = :test
+
+        get_as instructor, :create_groups_when_students_work_alone,
+               params: { course_id: course.id, assignment_id: @assignment.id },
+               format: 'js'
+
+        assert_enqueued_with(job: CreateGroupsJob, args: [@assignment, data])
+        expect(flash[:error]).to be_blank
+      end
+
+      it 'responds with _poll_job template' do
+        get_as instructor, :create_groups_when_students_work_alone,
+               params: { course_id: course.id, assignment_id: @assignment.id },
+               format: 'js'
+        expect(response).to render_template('shared/_poll_job')
+      end
+
+      it 'responds with appropriate status' do
+        get_as instructor, :create_groups_when_students_work_alone,
+               params: { course_id: course.id, assignment_id: @assignment.id },
+               format: 'js'
+        expect(response).to have_http_status(200)
+      end
+
+      it 'does not create groups if group_max > 1' do
```

Comment:
Then here, this `it` should be turned into a `context`, and you should repeat the same test structure as above within this `context` block, so that each `expect` is in a separate test case.